modules/apps/27-interchain-accounts/controller/ibc_middleware_test.go

```diff
@@ -960,22 +962,31 @@ func (suite *InterchainAccountsTestSuite) TestOnChanUpgradeAck() {
 func (suite *InterchainAccountsTestSuite) TestOnChanUpgradeOpen() {
 	var (
 		path                *ibctesting.Path
-		isNilApp            bool
 		counterpartyVersion string
 	)
 
 	testCases := []struct {
 		name     string
 		malleate func()
+		expPanic func() error
 	}{
 		{
-			"success",
+			"success", func() {}, nil,
+		},
+		{
+			"failure: upgrade route not found",
 			func() {},
+			func() error {
+				return errorsmod.Wrap(porttypes.ErrInvalidRoute, "upgrade route not found to module in application callstack")
+			},
 		},
 		{
-			"success: nil app",
```

Comment:
we should still keep this test case around :smile:

Reply:
I've reverted this, ty.